jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppClassLoader.java

```diff
@@ -325,10 +326,15 @@ private boolean isFileSupported(String file)
      */
     public void addJars(Resource lib)
     {
-        if (lib.exists() && lib.isDirectory())
+        if (lib.exists() && lib.isDirectory()) // TODO perhaps redundant given null check from list()?
```

Comment:
Javadoc is unclear, but typical implementations like `PathResource` would just swallow `NotDirectoryException` and return null later, which we need to check for anyway, so checking this in advance just seems like a waste of time.

Reply:
`addJars()` is a user exposed API.  its is often used from xml deployables and embedded-jetty instances.

Reply:
The Javadoc I was referring to was that of `exists`, `isDirectory`, and (especially) `list`.

Reply:
I'd like to keep this code as-is for jetty-9.4.x. We can consider other changes in jetty-10.

Reply:
> I'd like to keep this code as-is

Should I understand that to mean “minimize the diff here”? If so, can do.